### PR TITLE
pull s2 metadata from google 

### PR DESCRIPTION
**Pull request by jacquelynsmale**

### CODE REVIEW

File: tests/test_process.py [hunk 1]
```diff
@@ -129,50 +129,30 @@ def test_get_lc2_path():
...
-        status=404,
-    )
-    responses.add(
-        responses.POST, process.S2_SEARCH_URL,
-        body='{"numberReturned": 1, "features": [{"foo": "bar"}]}', status=200,
-    )
+def test_get_s2_path(test_data_directory):
+    scene_name = 'S2A_MSIL1C_20160616T112217_N0204_R137_T29QKF_20160617T193500'
+    f = open(f'{test_data_directory}/{scene_name}.manifest.safe', 'r')
+    manifest_text = f.read()
```
Comment (asjohnston-asf) [inline tests/test_process.py:146]: It's typical in Python to use `with` along with `open` so the file is automatically closed as soon as you're finished with it, e.g.:
```suggestion
    with open(f'{test_data_directory}/{scene_name}.manifest.safe', 'r') as f:
        manifest_text = f.read()
```

More details at https://towardsdatascience.com/why-you-should-use-context-managers-in-python-4f10fe231206

File: hyp3_autorift/process.py [hunk 2]
```diff
@@ -87,35 +86,57 @@ def get_lc2_path(metadata):
...
+        # post-2016-12-06 scene; only one tile
+        file_path = hrefs[0]
+    else:
+        # pre-2016-12-06 scene; choose the requested tile
+        tile_token = scene_name.split('_')[5]
+        file_path = [href for href in hrefs if href.endswith(f'_{tile_token}_B08.jp2')][0]
+    return f'/vsicurl/{S2_GRANULE_DIR}/{file_path}'
+
+
+def get_s2_bbox(path: str):
```
Comment (asjohnston-asf) [inline hyp3_autorift/process.py:104]: This isn't technically tied to Sentinel-2, so we could use a more generic name like `get_bbox_of_raster()`, even if we're only using it for Sentinel-2 at the moment. Thoughts?

File: tests/test_process.py [hunk 3]
```diff
@@ -129,50 +129,30 @@ def test_get_lc2_path():
...
-    responses.add(
-        responses.POST, process.S2_SEARCH_URL,
-        body='{"numberReturned": 0}', status=200,
-    )
-
-    s3_path = process.get_s2_metadata('S2B_60CWT_20220130_0_L1C')['assets']['B08']['href']
-    assert s3_path == 's3://sentinel-s2-l1c/tiles/60/C/WT/2022/1/30/0/B08.jp2'
+def test_get_s2_bbox(test_data_directory):
+    bbox = process.get_s2_bbox(str(test_data_directory / 'T60CWU_20160414T200612_B08.jp2'))
+    assert bbox == [-183.0008956, -78.4606571, -178.0958227, -77.438842]
```
Comment (asjohnston-asf) [inline tests/test_process.py:155]: This covers the antimeridian example; we can copy/paste a similar test for `T55XEE_20200911T034541_B08.jp2` to test a non-antimeridian case.

File: tests/test_process.py [hunk 4]
```diff
@@ -129,50 +129,30 @@ def test_get_lc2_path():
...
-        body='{"numberReturned": 1, "features": [{"foo": "bar"}]}', status=200,
-    )
+def test_get_s2_path(test_data_directory):
+    scene_name = 'S2A_MSIL1C_20160616T112217_N0204_R137_T29QKF_20160617T193500'
+    f = open(f'{test_data_directory}/{scene_name}.manifest.safe', 'r')
+    manifest_text = f.read()
+    path = process.get_s2_path(manifest_text, scene_name)
+    assert path == '/vsicurl/https://storage.googleapis.com/gcp-public-data-sentinel-2/tiles//./GRANULE' \
+                   '/S2A_OPER_MSI_L1C_TL_SGS__20160616T181414_A005139_T29QKF_N02.04/IMG_DATA' \
+                   '/S2A_OPER_MSI_L1C_TL_SGS__20160616T181414_A005139_T29QKF_B08.jp2'
```
Comment (asjohnston-asf) [inline tests/test_process.py:150]: This covers the pre-2016-12-06 case. Scenes after 2016-12-06 have a difference manifest.safe format, so we should copy/paste a similar test for `S2B_MSIL1C_20200419T060719_N0209_R105_T38EMQ_20200419T091056.manifest.safe` to test the more recent format.

File: hyp3_autorift/process.py (hunk 2 above)
Comment (jacquelynsmale) [reply hyp3_autorift/process.py:104]: That sounds good! I'll make that change

File: hyp3_autorift/process.py [hunk 5]
```diff
@@ -32,8 +32,7 @@
 gdal.UseExceptions()
 
 S3_CLIENT = boto3.client('s3')
-S2_SEARCH_URL = 'https://earth-search.aws.element84.com/v0/collections/sentinel-s2-l1c/items'
-S2_WEST_BUCKET = 's2-l1c-us-west-2'
+S2_GRANULE_DIR = 'https://storage.googleapis.com/gcp-public-data-sentinel-2/tiles/'
```
Comment (asjohnston-asf) [inline hyp3_autorift/process.py:35]: The `/` at the end here is leading to a double `//` in the manifest safe url generated on line 123, e.g.

`https://storage.googleapis.com/gcp-public-data-sentinel-2/tiles//38/E/MQ/S2B_MSIL1C_20200419T060719_N0209_R105_T38EMQ_20200419T091056.SAFE/manifest.safe` (generates an error)
`https://storage.googleapis.com/gcp-public-data-sentinel-2/tiles/38/E/MQ/S2B_MSIL1C_20200419T060719_N0209_R105_T38EMQ_20200419T091056.SAFE/manifest.safe` (successfully downloads the file)

I think dropping the trailing `/` is the easiest path forward? Or use something fancier like `os.path.join` or `urllib.parse.urljoin` to make it not matter whether there's an extra `/` or not?

```suggestion
S2_GRANULE_DIR = 'https://storage.googleapis.com/gcp-public-data-sentinel-2/tiles'
```

**Comment by asjohnston-asf** [comment]
Trufflehog failure is a false positive. The two manifest.safe files added to `hyp3_autorift/tests/data/` include file MD5 sums that are being flagged as high entropy.

**Review by asjohnston-asf — APPROVED** [review]

**Pull request merged**